docs/api/create_docker_container.md

```diff
@@ -0,0 +1,78 @@
+# Creating a container
+
+Testcontainers' generic container support offers the most flexibility, and makes it easy to use virtually any container images as temporary test environments. To interact or exchange data with a container, Testcontainers for .NET provides an implementation of the `TestcontainersBuilder<TDockerContainer>` to configure and create the resource.
+
+## Examples
+
+A NGINX container that binds the HTTP port and hosts static content. Hit the static content in your browser at `http://localhost:80` or `http://host-ip:80`.
+
+```csharp
+await new TestcontainersBuilder<TestcontainersContainer>()
+  .WithName(Guid.NewGuid().ToString("D"))
+  .WithImage("nginx")
+  .WithPortBinding(80)
+  .Build()
+  .StartAsync()
+  .ConfigureAwait(false);
+```
+
+Listens on port `80` and always returns `42`. Connect via `nc`, `telnet` or a `Socket` implementation to `localhost:80` or `host-ip:80`.
+
+```csharp
+await new TestcontainersBuilder<TestcontainersContainer>()
+  .WithName(Guid.NewGuid().ToString("D"))
+  .WithImage("alpine")
+  .WithExposedPort(80)
+  .WithPortBinding(80)
+  .WithEnvironment("MAGIC_NUMBER", "42")
+  .WithEntrypoint("/bin/sh", "-c")
+  .WithCommand("while true; do echo \"$MAGIC_NUMBER\" | nc -l -p 80; done")
+  .Build()
+  .StartAsync()
+  .ConfigureAwait(false);
+```
+
+!!!tip
+
+    To avoid port conflicts, do not bind a fix host port. Instead, assign a random host port by using `WithPortBinding(80, true)` and retrieve it from the container instance by using `GetMappedPublicPort(80)`.
+
+## Supported commands
+
+| Builder method                          | Description                                                                                                     |
+|-----------------------------------------|-----------------------------------------------------------------------------------------------------------------|
+| `WithDockerEndpoint`                    | Sets the Docker daemon socket to connect to.                                                                    |
+| `WithAutoRemove`                        | Will remove the stopped container automatically like `--rm`.                                                    |
+| `WithCleanUp`                           | Will remove the container automatically after all tests have been run.                                          |
+| `WithLabel`                             | Applies metadata to the container e.g. `-l, --label testcontainers=awesome`.                                    |
```

Comment:
```suggestion
| `WithLabel`                             | Applies metadata to the container e.g. `--label testcontainers=awesome`.                                    |
```

Reply:
If added to each method the equivalent CLI arg, some of them have a long and short version. Why should we remove it?